### DISCUSSION TITLE
Starting back from scratch - Keypads / Scenes

### DISCUSSION THREAD

**Discussion by JimCSaratoga** [kuestess/homebridge-platform-insteonlocal#240] [Q&A]
Apologies if this is discussed elsewhere but I'm trying to clean up my install and have found that there are still schedules in my hub (pre-Insteon server shutdown hub) that are interfering with my InsteonLocal configuration.  If I were to start with a new hub, factory reset all of the switches, keypads, plug-ins, motion sensors, etc I am comfortable using the {IP address of homebridge}:3000 interface however do not understand clearly how to create a scene that involves a keypad button.  @kuestess I read your group = keypad button letter (For a keypad, the group number corresponds to the button number (ie, A=1, B=2, C=3, etc)) but am not clear on how the main group number (for example picking 20 for the group number) would work vs. the keypad group number.   If anyone has had success could you kindly post screenshots using a keypad button (or even two would be great) and some switches to a hub using the :3000 web interface?  Thanks for any and all information in advance.  Jim

**Comment by sahafeez** [comment]
Do not wipe your hub. There are many reports of it never recovering. 

You can go in an remove anything you see in the Express server. 

http://yourhost:3000/hub

Pull the links. You can see the scene also there. You can see the scene ID and then just delete everything from the links.

**Comment by JimCSaratoga** [comment]
Thanks @sahafeez  - I appreciate the advice on the hub.  Switching gears - any thoughts on creating a new scene as described above?  I've tried multiple times (on my test hub / hombridge) and the 'create scene' button press never completes.  @kuestess any thoughts?  I've even tried making a simple scene with just the hub and a 2477S.

**Comment by 64Spaces** [comment]
Greetings. Thank you for starting this discussion. I don’t have an answer to your linking question. But I’m also considering a fresh install and have a question. Is it enough to delete the insteon.json file and the persist and accessories files? Or did you delete other things? Thank you, Ruben

**Reply by JimCSaratoga** [reply]
Hi Ruben - the Insteon.json file exists in the file bundle of the InsteonLocal plug-in bundle.  It's a JSON configuration of the configuration you have been building with the InsteonLocal plug-in (it records the data pulled from the Insteon hub).  

That said if you do want to start from scratch I suggest that you go on eBay and buy a used 2245 hub (2017 or newer - look for the last two digits on the information sticker on the back of the hub - e.g. rev 2.8 - 3417 (means it was manufactured in 2017 so newer sourced capacitor and board change for heat - the newer hubs have a better chance of being factory reset while preserving your current working hub).  

At that point you can link each of your devices with the new hub (yes it will add additional links into each device but I've been doing that for years without an issue).  To link you would use the last tab ("link") on the https://{ip address of the newly installed hoobs, home bridge on RPi, home bridge on Mac, windows, etc}:3000 - e.g. https://192.168.1.20:3000  and then in the top box (you can do multiple at once) start putting in the 6 hex character serial numbers of the switches, dimmers, etc to pair the hub. you will hear the pairing beep just like in the good old days when the Insteon apps worked.  

That brings us to where I started this thread which is not being able to use the lower boxes on that same "link" screen to create scenes.  I've found that the Insteon scenes are still far more reliable vs. creating them in Apple Home and making a switch for that scene ie. { "name": "outside lights", "groupID": 25, "deviceType": "scene", "dimmable": "no", "groupMembers" etc. etc }

if anyone has been able to make New scenes can you please post a video ?  Maybe I'm not putting in the correct parameters for ramp rate, etc. - it just doesn't work for me.

**Comment by sahafeez** [comment]
@JimCSaratoga @64Spaces 

So I am not sure what is understood so let me start from the top.

config.json is configuration file for homebridge. Any chances you make in the express server that have to do with link/unlink, scenes, etc. are not reflected there. If you delete a device in express you need to remove it in config.json.

insteon.json is what the express server THINKS the insteon DB on the HUB/PLM looks like. There are a few issues with the express server from what I have seen:

1. get all Links never really works correctly - it can and will failout and then tell you it got everything.
2. get all links on a device will report no changes and ask you to fetch anyway, and if you do you will find links, etc. that express did not know about.
3. the creation of a scene or anything else that puts up a popup status dialog may sometimes not return

Solutions:

1-2 - you need to go to EACH and every device and click get all links and yes to the nothing changed. if you do this for every device then insteon.json will reflect the true state of things and they it is simpler to understand why a things are happening. make sure to click save every once in a while in the device as I have seen it crash.

3. have the homebridge console window open looking at the logs and then submit the scene creation. you should see the task of the express server logged. I think it will say scene created for every item you include. You should be able to tell when it is done, even if the dialog box is still up in express. okay to click away on that after the logs say done.

@kuestess all the issues above are reproducible and I have opened a few issues on them. I know you have a huge influx of new people now and this is not your day job. Let me know if there is anything I can do to help (learning js is on the list at some point).

**Reply by kuestess** [reply]
Thanks @sahafeez - get all links likely fails because there’s a battery operated device that doesn’t respond, so I likely just need better error handling.  The get device links looks at the database delta compared to what it gets back from the device and pops up that warning when they match (meaning the database hasn’t changed) so it could be your devices lying to you :-) or something I need to address:-). I haven’t had any issues with scene creation, so if you have a log that points to the failure that would be helpful.

**Reply by JimCSaratoga** [reply]
@sahafeez that was very helpful thank you !!! - especially #3 I see the "no return" constantly

@kuestess also very helpful - I do have two motion sensors (the original ones from Insteon) so maybe that's why I'm having an issue - I'll turn on debug (migrating FINALLY everything over to my hombridge running on my Synology vs. dealing with mission impossible: getting hoobs to go into debug mode.  I should be able to get the debug log then

You ALL ROCK !!!

**Comment by sahafeez** [comment]
@kuestess I am aware of the battery issue. When I moved things over to the USB from the HUB I only moved non-battery items to start and it still would not always work. I have the battery items including now so no going back.

I think the no change, oh no there are links is a function of the first get all links not completing.

1. get all links 
2. fails out but devices that did not get pulled get marked as pulled
3. go look at device, see NO links, click get, say no change, click yes, links.

A feature like this would be cool.

Force get where:
1. it will handle NULL reply and move on
2. it will not care about the device DB vs insteon.json status and will just get the links
3. in the case of 2 if insteon has links and the get errors out it should not overwrite what is currently there.

**Comment by sahafeez** [comment]
@kuestess to be clear the scene is created just fine. the issue is that dialog that pops up when you hit submit never goes aways. the logs show it completes. you can then navigate away. running safari latest ga on latest macOS ga.

**Reply by 64Spaces** [reply]
Thank you both. Actually, after reading somewhere that Scott uses a PLM, I’m trying to load everything on a new PLM. So I want to delete my Hub Plugin and reload it on Homebridge using the PLM. I’m assuming I can delete the json files so there’s no remnants of the devices in Homebridge. I have Mosquito Broker and a Zigbee MQTT plug-in that are all set up and I don’t want to redo these. So after getting the PLM set up, I’m going to give it a try. Unless you have a different suggestion?

**Reply by kuestess** [reply]
Thanks @sahafeez that helps narrow down the issue - I'll get a fix into the next update @64Spaces Yep, if you delete the Insteon portion of your config.json and the Insteon.json, it'll be like a brand new install.
